compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -330,6 +359,62 @@ fn gather_explicit_predicates_of(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::Gen
     }
 }
 
+fn bidirectional_lifetime_predicates<'tcx>(
```

Comment:
Can you get rid of this function, and just change the one callsite of `bidirectional_lifetime_predicates` to just call `generics_of` there and pass in a `&mut Vec`?

Reply:
Please just make one version of this function